File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/debug/blockchainInspection/DebugStartCollectingTrieStatsTest.java
package opensdk.sdk.apis.debug.blockchainInspection;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugStartCollectingTrieStatsResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugStartCollectingTrieStatsTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_startCollectingTrieStats")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String address = "0x0000000000000000000000000000000000000000";

        DebugStartCollectingTrieStatsResponse response = sdk.debug.startCollectingTrieStats(address).send();

        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/blockchainInspection/DebugStartCollectingTrieStats.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk.DebugApi(new OpenSdk.ApiClient(RPC));

describe('debug_startCollectingTrieStats API', () => {
    test('should return debug_startCollectingTrieStats', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(data).toBe(null);
            done();
        };

        const address = "0x0000000000000000000000000000000000000000";

        sdk.startCollectingTrieStats(address, {}, callbackOne);
    });
});

